tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,247 @@
+$existingColumns = $db->pselect($columnQuery, array());
+
+// Format result to only keep consent name columns
+$formattedColumns = [];
+foreach ($existingColumns as $column) {
+    $columnName = $column['Column_name']; 
+    if (!(preg_match("/date/", $columnName) || preg_match("/withdrawal/", $columnName))) {
+        array_push($formattedColumns, $columnName);
+    }
+}
+// Check that column names are in list of consent names
+foreach ($formattedColumns as $columnName) {
+    $i=0;
+    foreach ($consents as $consent){
+        $consentName = $consent['name'];
+        if($consentName===$columnName) {
+            $i++;
+        }
+    }
+    if ($i===0) {
+        array_push(
+            $errors, "The consent type " . $columnName . " exists in the database but not in Config.xml.
+           Please add the consent to Config.xml or delete columns and data from 'participant_status'"
+        );
+    }
+}
+// Check that consent status, date, and withdrawal columns exist in participant_status
+foreach ($consents as $consent) {
+    $consentName       = $consent['name'];
+    $consentDate       = $consentName . "_date";
+    $consentWithdrawal = $consentName . "_withdrawal";
+    $statusExists      = $db->columnExists('participant_status', $consentName);
+    $dateExists        = $db->columnExists('participant_status', $consentDate);
+    $withdrawalExists  = $db->columnExists('participant_status', $consentWithdrawal);
+    if (!($statusExists && $dateExists && $withdrawalExists)) {
+        array_push($errors, "At least one column is missing for " . $consentName . " in participant_status table.
+           Check that status, date, and withdrawal columns exist for this consent type.");
+    } else {
+        // Check for zero dates
+        $dateQuery = 'SELECT ID, CandID, ' . $consentDate . ', ' . $consentWithdrawal . ' FROM participant_status 
+                      WHERE ' . $consentName . ' IS NOT NULL OR ' . $consentName . ' != ""';
+        $psData    = $db->pselect($dateQuery, array());
+        foreach ($psData as $entry) {
+            if($entry[$consentDate] === "0000-00-00" || $entry[$consentWithdrawal] === "0000-00-00") {
+                array_push($errors, "Zero dates found in participant_status for:
+                           [ID]     => " . $entry['ID'] . "
+                           [CandID] => " . $entry['CandID'] . "
+                           Please remove date or run /tools/DB_date_zeros_removal.php.");
+            }
+        }
+    }
+}
+//////////////////
+// THROW ERRORS //
+//////////////////
+if (!empty($errors)) {
+    print_r($errors);
+    die("Resolve errors and run script again.\n");
+} else {
+    echo "\nValidation successful.\n";
+}
+
+/////////////////////
+// CONTINUE SCRIPT //
+/////////////////////
+
+// Update ConfigSetting table with value of 'useConsent' if true. Default is set to false.
+$configID = $db->pselectOne(
+    'SELECT ID FROM ConfigSettings WHERE Name="useConsent"',
```

Comment:
quotes are backwards.. `'` is the standard way to quote strings in SQL, `"` just happens to work in MySQL